tawazi/_dag/dag.py

```diff
@@ -529,13 +482,22 @@ def __call__(self, *args: P.args, **kwargs: P.kwargs) -> RVDAG:
         """
         if kwargs:
             raise TawaziUsageError(f"currently DAG does not support keyword arguments: {kwargs}")
-        # 1. generate the subgraph to be executed
-        graph = self.make_subgraph()
 
-        # 2. copy the ExecNodes
-        call_xn_dict = self.make_call_xn_dict(*args)
+        graph = deepcopy(self.graph_ids)
+
+        # add debug nodes
+        if cfg.RUN_DEBUG_NODES:
+            debug_ids = self.graph_ids.include_debug_nodes(graph.leaf_nodes)
+            graph = self.graph_ids.subgraph(list(graph.nodes) + debug_ids).copy()
+
+        # 3. Remove debug nodes
+        else:
+            graph.remove_nodes_from([node_id for node_id in graph.debug_nodes])
```

Comment:
why remove them ?
Are they existing by default or not ?
Either, by default they aren't part of the graph, and we add them optionally (which is more optimal)
OR, by default they are part of the graph, and we remove them optionally (which is not optimal)

Reply:
so that was the "old" way of doing things, tbh I haven't refactored this part, this PR  being exclusively on trying to remove from the dag interface the `make_subgraph` method. I'm also not really a fan of the way we "discover" debug nodes since it's a hassle to have to go through the process of recomputing everything every call. I clearly think this could be improved, I don't have a definite idea yet, do you want me to try to do that in this PR ?

Reply:
no, it's the same as it was before this PR : the original graph has all nodes, and when we build it, either we're in debug mode or we're not, in which case we prune all debug nodes

Reply:
@bashirmindee this was refactored you can have a look